### ISSUE TITLE
New user informative e-mail, with instructions to set first time password

### ISSUE THREAD

**Issue by erikgeurts** [revive-adserver/revive-adserver#109]
Currently, when a new user is created, their username, full name, e-mail address and password need to be entered manually, and then sent to that user by e-mail or some other method.

I would like to propose a few changes and enhancements, as follows:
- creating a new user will only require entering an e-mail address and a username, entering a full name is optional
- when a new username is created, the software sends them a "Welcome new user" type of e-mail
- meanwhile, the new user remains 'disabled' as their username has not been confirmed yet
- the e-mail message contains a link that will send them to a web page of Revive Adserver, that is in essence the 'forgot password' feature which already exists
- on that page, the new user is required to enter a password for their username, and also to confirm their full name (or correct it)
- the page will also include a link or button that will generate a random and safe password, so that the new user does not have to think up a password
- only after the new user enters and saves the password and their full name, their username will be enabled
- if the new user does not confirm their username within a certain time frame (parameter), the new user's record will be deleted automatically

The advantages of this method are:
- the person creating the new user does not have to know the new user's password
- the new password is not transmitted via e-mail to the new user
- the e-mail message that is used to send the new user their personal 'set your password' link also acts as an check to make sure the new user's e-mail address was entered correctly

**Comment by mbeccati** [comment]
Counter-proposal to reduce the number of new features requiring development, ensuring a deliverable for 5.4:

* User creation same as today, without the password.
* The user database record will have an invalid password hash, hence it is practically disabled
* The user will receive the "Welcome new user" email
* The email contains a link to the password recovery wizard
* Password suggestion is left to the browser or password managers
* The link follows the same rules as other password recoveries
* No automatic cleanup

**Comment by erikgeurts** [comment]
> Counter-proposal to reduce the number of new features requiring development, ensuring a deliverable for 5.4:
> 
> * User creation same as today, without the password.
> * The user database record will have an invalid password hash, hence it is practically disabled
> * The user will receive the "Welcome new user" email
> * The email contains a link to the password recovery wizard
> * Password suggestion is left to the browser or password managers
> * The link follows the same rules as other password recoveries
> * No automatic cleanup

This appears to satisfy most of the important parts of what I proposed:
- user A doesn't enter and therefor know the password of new user B, and thus doesn't have to send that password to user B using some unsafe method
- user B gets a single email, welcoming them, and presenting a link that will take them to the web for where then can enter their first password (#109).
- User B will be encouraged to enter a safe password because of the password strength indicator that's part of #1291 

We do have to check and make sure that browsers and password managers recognize that the password field is indeed a password field. In most web applications I'm familiar with, Chrome (for example) automatically recognizes certain fields as being password fields and then automatically suggests a safe password. However, in Revive Adserver it does not do that. Perhaps there is some sort of naming convention, extra label or extra attribute on the password field that helps with that recognition? I realize this is a separate topic, and I'm more than happy to open a new issue for it. But I also believe it is probably fairly simple to do, so hopefully it can be included in the upcoming milestone release.

Perhaps we can include the remaining elements of my idea (like automatic cleanup and generating a safe password) in a future release.
